### PR TITLE
Avoid negative strides for tensors

### PR DESCRIPTION
**Pull request by shirayu**
The original lines cause the following error for tensors.
```
At least one stride in the given numpy array is negative, and tensors with negative strides are not currently supported. (You can probably work around this by making a copy of your array  with array.copy().) 
```

The same operation is done here.

https://github.com/huggingface/diffusers/blob/f1484b81b0b763aa010fd2083df933305afb1812/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L253-L258

(Probably related to #551 )

### CODE REVIEW

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py [hunk 1]
```diff
@@ -284,7 +284,10 @@ def __call__(
 
         # Some schedulers like PNDM have timesteps as arrays
         # It's more optimzed to move all timesteps to correct device beforehand
-        timesteps_tensor = torch.tensor(self.scheduler.timesteps[t_start:], device=self.device)
+        if torch.is_tensor(self.scheduler.timesteps):
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py:287]: `self.scheduler.timesteps` should always be a tensor - I don't think we should have these if statements here (netiher in `pipeline_stable_diffusion.py` BTW)

@kashif @anton-l could we make sure that `self.scheduler.timesteps` is always of type `torch`

Comment (kashif) [reply src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py:287]: ok... since its always used for the tqdm counter I never converted it to torch in the torch-only PR... should i make it always a torch tensor?

**Comment by kashif** [comment]
@shirayu my feeling was that the imge2img pipeline would work without the need to convert the scheduler's timesteps to a torch tensor. If you have a look, the `unet` can take the time index as `int` and the `callback` also expects the time index to be an `int`. Can you kindly confirm if this pipeline works without the need to `torch.tensor` the `timesteps`?

**Comment by shirayu** [comment]
@kashif  Thank you for the comment. 
I updated them (7f567ec) and works fine.

**Comment by kashif** [comment]
while you are there you could fix the spelling of optimzed -> optimized and I believe the same would hold for the stable diffusion too? what would you say @anton-l ?

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py (hunk 1 above)
Comment (patrickvonplaten) [reply src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py]: Ah yes I think we should always make it a PyTorch tensor now that schedulers are PyTorch-only . Think `tqdm` should be a 2nd citizen here :-)

Comment (kashif) [reply src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py]: I see... the PR now removes the need for the move to pytorch tensor and all seems to work too... but yes all torch tensors would also work similarly then...

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py [hunk 2]
```diff
@@ -315,12 +315,9 @@ def __call__(
         latents = init_latents
 
         t_start = max(num_inference_steps - init_timestep + offset, 0)
+        timesteps = self.scheduler.timesteps.copy()[t_start:]
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py:318]: why do we need a `copy()` here?

Comment (shirayu) [reply src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py]: Oh, It is just what I forgot to remove and is a needless copy.
I removed it at dca59b8, but also works fine.

**Comment by patrickvonplaten** [comment]
Tests all pass -> merging!
